### PR TITLE
New wizard spell: Pain Mirror

### PR DESCRIPTION
**Pull request by unid15**
- Adds Pain Mirror: a spell with cooldown of 90 (every cooldown upgrade takes off 20 seconds) and duration of 5 seconds. While active, all damage that you receive will be also dealt to everybody around you
- Activating the spell plays the vampire intro sound, and while it's active you're surrounded by a purple overlay. There is a visual effect displayed when you reflect damage.
- You always reflect the same amount of damage that you receive, after all multipliers. Diona wizards hit by a 40 damage laser beam will deal 80 damage to everybody around them (but then die anyways :cry:)
- A single instance of damage coming from this spell can't deal more than 80 damage. This means that suiciding (which gives you 200 damage) will only deal 80 damage to everybody nearby.

graf
- After the round ends, you can see how effectively the spell was used (amount of damage taken with the spell active, and amount of damage dealt by the spell are shown near the spell itself)

TODO:
- [x] Make the messages better
- [x] Balance
- [x] Test stuff

### CODE REVIEW

**Comment by chatsonf** [comment]
Now we need a heal spell so wizards aren't doomed when they get knocked down

File: code/modules/spells/general/reflect_pain.dm [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+/spell/mirror_of_pain/before_cast()
+	return
+
+/spell/mirror_of_pain/choose_targets(mob/user = usr)
+	return list(user)
+
+/spell/mirror_of_pain/cast(list/targets, mob/user)
+	for(var/mob/living/L in targets)
+		to_chat(L, "<span class='sinister'>Your life essence .</span>")
```
Comment (ComicIronic) [inline code/modules/spells/general/reflect_pain.dm]: MY life essence?

Comment (Exxion) [reply code/modules/spells/general/reflect_pain.dm]: > Space before the period

**Comment by Exxion** [comment]
HOW 2 WIN WIZARD WARS:
1. Go diona
2. 1 Jaunt, 3 Pain Mirror, 1 Mutate
3. Steal fire axe
4. Hulk out
5. Activate pain mirror
6. Run around bashing yourself with fire axe

But since dionae can't be wizards under normal circumstances, :+1:

**Comment by AI-Is-Rogue** [comment]
@Exxion Just hope there is a Diona on station to mindswap into.

**Comment by ihadtoregisterforthis** [comment]
And grab a light too, because the damage dealt will be as small as the damage you recieve.

**Comment by jknpj** [comment]
does it work in simple mobs?

**Comment by Kurfursten** [comment]
I'm not sure I like this from a game design standpoint. Most of players don't know a **lot** about the game in general, and especially antagonists. Recently someone added an information message to coffins because players weren't realizing they had to re-enter their bodies to get the green lights to appear. That's something I consider a very base, entry-level piece of medbay info.

I think this means that ~~Malachi~~ a new player is probably going to wipe out the sec team because they keep attacking the wizard "when he goes supersaiyan and rings the bell". Basically, you're punishing _the entire crew_ because one person didn't learn counterplay to an antagonist. It's not intuitive either - glowing red is not immediately apparent the first time you see it of "don't attack this" - for instance, animated items have a red "glow" effect.

It's like the spell equivalent of encouraging players to welderbomb/flamethrowers on the blob.

**Comment by GenericGoose** [comment]
@jknpj it works on all mob/living
so yea

**Comment by unid15** [comment]
@Kurfursten I'll try to make the visual effects more obvious, but IMO the actual effect is more or less balanced - the spell won't kill everybody around you unless you die yourself (or somebody was already injured). And it won't prevent chloral or internal bleeding from a shotgun

This might have to be rebalanced once we get a healing spell though

**Comment by Kurfursten** [comment]
I could definitely see it getting out of hand with a healing spell, especially since the wizard might roll nobleed after 20 brute but everyone else does roll a bleed.

**Comment by ghost** [comment]
Could the wizard type suicide then activate the spell, thus putting everyone around him into crit?

**Comment by Blithering** [comment]
Whats the range on the spell?

**Comment by unid15** [comment]
The range is 7, as in the default view. Suicide currently works, but I'll probably cap the damage at something like 30-40 to prevent that

**Comment by ghost** [comment]
@unid15 why though? If a wizard commits suicide to kill a bunch of others I think that's great rather than something to avoid.

**Comment by Exxion** [comment]
Yeah, let it happen
1. That would unrevivably kill the wizard (not that wizards can be revived in any roundtype in which they naturally spawn anyway)
2. It would kill everyone in view with pure oxyloss damage, meaning that they could be revived to full health with just a defib

**Comment by ComicIronic** [comment]
Not all suicides are oxyloss. I think if a wiz suicided by a burnloss method, he could maybe husk the corpses.

**Comment by chatsonf** [comment]
Point 1 still stands, since if the wizard kills himself then the round is generally over.

**Comment by SonixApache** [comment]
If the wizard gets hit by something that makes him gib, does it gib everyone within the spell range as well?

**Comment by unid15** [comment]
Suicide wizards might get nasty in ragin' mages and bussed wizards though
@SonixApache nope, instant death like gibbing is not reflected, only damage is

Going to continue working on this after the spellbook refactor is in (as it allows some things to be done easier)

**Comment by Shadowmech88** [comment]
I'd recommend adding some kind of _very clear_ visual effect, branching from the wizard to his victims, to illustrate that they are taking damage from the wizard.

Like one of those soulstone particle effects or an energy bolt or a wave that spreads outward from the wizard or something.

**Comment by unid15** [comment]
test

**Pull request merged**